lib/kill.js

```diff
@@ -51,27 +51,22 @@ export const spawnedCancel = (spawned, context) => {
 	}
 };
 
-const killAfterTimeout = async (spawned, timeout, killSignal, controller) => {
+const killAfterTimeout = async ({spawned, timeout, killSignal, context, controller}) => {
 	await pSetTimeout(timeout, undefined, {ref: false, signal: controller.signal});
 	spawned.kill(killSignal);
-	throw Object.assign(new Error('Timed out'), {timedOut: true, signal: killSignal});
+	Object.assign(context, {timedOut: true, signal: killSignal});
```

Comment:
Pass this information using the `context` object instead of attaching it to the thrown error. 
Like this, if some library adds some unrelated `error.signal` property, we will ignore it.
The final error thrown by Execa uses `makeError()` which re-adds those properties.